test/cli_test/test_pcap.py

```diff
@@ -68,39 +53,43 @@ def test_pcap_file_write_features_selection(self):
-                        '--save-features-format', output_format,
-                        '--tcp',
-                        '--ipv4',
-                        '--aggregator', 'index',
-                        '--label-file', TEST_DATA / 'single-pcap' / 'labels.txt',
-                        '--pcap-file', TEST_DATA / 'single-pcap' / 'test.pcap',
-                        '--output', temp_path,
-                        '--quiet',  # autogluon's threading makes capturing/suppressing
-                                    # its stdout a little harder
-                    )
-
-                    npt_dir = temp_path / 'nprint'
-                    self.assertFalse(npt_dir.exists())
-
-                    format_scheme = output_format.split('.')[-1]
-                    feature_path = temp_path / 'feature'
-                    self.assertTrue(any(feature_path.glob(f'features*.{format_scheme}')))
-
-                    graphs_path = temp_path / 'model' / 'graphs'
-                    self.assertTrue(any(graphs_path.glob('*.pdf')))
-
-                    models_path = temp_path / 'model' / 'models'
-                    self.assertTrue(any(models_path.rglob('*.pkl')))
-
-                    meta_path = temp_path / 'meta.toml'
-                    self.assertTrue(meta_path.exists())
-                    self.assertEqual(toml.load(meta_path).get('nprint', {}).get('cmd'),
-                                     'nprint --pcap_file [input_pcap] --ipv4 --tcp')
+            with self.subTest(output_format=output_format), \
+                 no_learn_patch as no_learn_mock, \
+                 mktestdir() as tempdir:
+
+                temp_path = pathlib.Path(tempdir)
+
+                self.try_execute(
+                    '--save-features-format', output_format,
+                    '--tcp',
+                    '--ipv4',
+                    '--aggregator', 'index',
+                    '--label-file', TEST_DATA / 'single-pcap' / 'labels.txt',
+                    '--pcap-file', TEST_DATA / 'single-pcap' / 'test.pcap',
+                    '--output', temp_path,
+                    '--quiet',  # autogluon's threading makes capturing/suppressing
+                                # its stdout a little harder
+                )
+
+                npt_dir = temp_path / 'nprint'
+                self.assertFalse(npt_dir.exists())
+
+                feature_path = temp_path / 'feature'
+                format_id = output_format[0]  # c, p or f
+                feature_path_stream = feature_path.glob(f'features.{format_id}*')
+                self.assertEqual(sum(1 for file_ in feature_path_stream), 1)
+
+                # note: no model *artifact* assertions -- learn disabled
+
+                no_learn_mock.assert_called_once()
+                (learn_call,) = no_learn_mock.mock_calls
+                (_args, results) = learn_call.args if sys.version_info >= (3, 8) else learn_call[1]
+                self.assertSequenceEqual(results.features.shape, (4921, 961))
```

Comment:
only functional change to this test is that it no longer actually runs AutoML – (we already run AutoML *a lot* in tests) – rather, it saves time by mocking out that step and just checking that it was invoked with the expected parameters. (and, the point of this test is just to check that features are written in the requested format, which it asserts, above.)